src/Build.UnitTests/BackEnd/BuildRequestEngine_Tests.cs

```diff
@@ -342,9 +342,15 @@ public void TestEngineShutdownWhileActive()
 
             VerifyEngineStatus(BuildRequestEngineStatus.Uninitialized);
             _engine.InitializeForBuild(new NodeLoggingContext(_host.LoggingService, 0, false));
+            // We neeed to get the status changed AutoResetEvent returned to the non-signaled state correctly for verifying the engine status via waiting for a signal next time.
+            // Make sure the status change finishes
+            VerifyEngineStatus(BuildRequestEngineStatus.Idle);
+            // Make sure the status changed AutoResetEvent is reset in case inside VerifyEngineStatus method it doesn't go to call WaitForEvent method.
+            _engineStatusChangedEvent.Reset();
+
             _engine.SubmitBuildRequest(request);
-            Thread.Sleep(250);
             VerifyEngineStatus(BuildRequestEngineStatus.Active);
+            _engineStatusChangedEvent.Reset();
```

Comment:
Sometimes the status changes so fast that the status is already the expected one for `VerifyEngineStatus`. Then inside `VerifyEngineStatus` it doesn't execute waitone by the waithandle, neither reset the signal state (see below). Next wait may be given with the signal set by previous status change. So explicit reset was added.

https://github.com/dotnet/msbuild/blob/1b84c9b5c6d264bab83114c6749def66d5656e6b/src/Build.UnitTests/BackEnd/BuildRequestEngine_Tests.cs#L516-L521
After a second thought, probably it's better to let the waithandle waitone after each status change by removing the following snippet from the above, rather than explicit reset.
```
 if (engine.Status == expectedStatus) 
 { 
     return; 
 } 
```